agentos/repo.py

```diff
@@ -62,6 +67,7 @@ class GitHubRepo(Repo):
     def __init__(self, name: str, url: str):
         self.name = name
         self.type = RepoType.GITHUB
+        url = url.replace("git@github.com:", "https://github.com/")
```

Comment:
Why do we enforce this? Does Dulwich only work with HTTPS and not ssh?

Reply:
The difference, I believe, is the auth flow that git uses.  On my machine:
* Cloning a `git@github.com` repo makes me enter the password on the key I use to auth with github (even though I'm just trying to read and not write the repo).  The clone process just hangs until I unlock the key.
* A `https://github.com/` repo, on the other hand, can clone a repo without any user input.

I'm not sure if this is universally true or just particular to my setup.  For now I'll add a comment to note this and keep a mental note to make the git interaction process more robust overall.